src/snowflake/snowpark/dataframe.py

```diff
@@ -1541,7 +1556,14 @@ def select(
 
         for e in exprs:
             if isinstance(e, Column):
-                names.append(e._named())
+                if self._all_variant_cols:
+                    names.append(
+                        Column(
+                            e._expr1, e._expr2, e._ast, _is_qualified_name=True
+                        )._named()
+                    )
+                else:
+                    names.append(e._named())
```

Comment:
can you help me remind the details -- do we parse xml and store content as object in VariantType column?

Reply:
yes